### ISSUE TITLE
Regression in vget/vset --format=json

### ISSUE THREAD

**Issue by jonhattan** [drush-ops/drush#499]
The output of vget has changed after the adoption of outputformats engine. This affectes scripted solutions since drush has lost the ability to vset a vgetted value.

This is drush 5 output:

```bash
$ drush @dr vget node_options_event --format=json
["status","promote"]
```

Drush 6:

```bash
$ drush @dr vget node_options_event --format=json
{"node_options_event":["status","promote"]}
```

In drush 5 you could pipe a vgetted value to vset:

```bash
$ drush @dr vget node_options_event --format=json | drush @dr vset node_options_event --format=json -
node_options_event was set to                                                                                                                                                                           [success]
array (
  0 => 'status',
  1 => 'promote',
).
```

In drush6, extra nesting is added:

```bash
$ drush @dr vget node_options_event --format=json | drush @dr vset node_options_event --format=json -
node_options_event was set to                                                                                                                                                                           [success]
array (
  'node_options_event' => 
  array (
    0 => 'status',
    1 => 'promote',
  ),
).
```

OTOH, drush 6 behaviour is consistent when several variables are returned, while drush 5 is useless.

There're two possible resolutions:

 * Change vget to return the value as drush5 when --exact is used.
 * Extend vset to consider input data as multi-variable when `-` is used as variable name.

I prefer option 1.

**Comment by greg-1-anderson** [comment]
After fiddling with this a bit, and settled on option 1.  Option 3 was tempting, but it is bad for scripts when you cannot predict the format of your output data.

Maybe we should reverse the default value for --exact, and require --exact=0 when multiple values are desired.  We could also force --exact=0 when no args are provided ("give me everything").  This would simplify the cli usage in the typical cases.

**Comment by greg-1-anderson** [comment]
Okay, that PR is no good.  The problem is that the default output format for vget is 'labeled export', and this formatter requires an associative array of key / value pairs, like the current default output data structure.  If you simplify the output structure with --exact, then you don't get any output.

SO.

Maybe we need to have a separate option that the user can set to simplify the output?  This option would force --exact, and would fail if the output format was something like 'labeled export'.  Meh.

Thoughts?  We could change the default format of vget to yaml.  That looks a lot like labeled export, and would continue to work if we simplified the output structure based on --exact.  Maybe this is best.  Folks could still manually use --format=labeled-export if they don't use --exact.  Meh -- still no way to use labeled export with --exact, which makes the labeled export format not-so useful.  This format is basically for Drush 5 consistency, so perhaps we can just forget about it for vget.

**Comment by greg-1-anderson** [comment]
Okay, I updated the PR, and all of the tests are passing now.  Here's what I did:

1. No more --format=labeled-export for vget.  The default format is now yaml, which looks very similar to the former format.  The default format for --pipe is still 'config'.
2. In --exact mode, the output of vget is trasformed to to be like Drush 5.
3. In --exact --pipe mode, the output of vget is NOT transformed.

As an added bonus, I also supported --format=yaml for vset.

That means that this still works:

```$ drush @dr vget node_options_event --format=json | drush @dr vset node_options_event --format=json -```

... and this also works:

```drush @dr vget node_options_event | drush @dr vset node_options_event --format=yaml -```

If these changes look good, I'll merge the branch.
